c10/util/ParallelGuard.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+
+namespace c10 {
+
+// Returns true if executed while a ParallelGuard object is alive in the current
+// thread.
+C10_API bool is_parallel_guard_alive();
```

Comment:
nit: if we want this to be closer to GradMode and other similar flags, you can move this to `ParallelGuard::is_enabled()`.